pkgs/applications/networking/cloudflared/default.nix

```
@@ -1,22 +1,35 @@
-{ lib, buildGoModule, fetchFromGitHub }:
+{ lib, buildGoModule, fetchFromGitHub, stdenv }:
 
 buildGoModule rec {
   pname = "cloudflared";
-  version = "2022.4.0";
+  version = "2022.4.1";
 
   src = fetchFromGitHub {
     owner  = "cloudflare";
     repo   = "cloudflared";
     rev    = version;
-    hash   = "sha256-+40OK2q4WdvlLhoPfZH6q+pghgS7ZLmaZl2VbZK4rdA=";
+    hash   = "sha256-dgvXbWtLP6sXBlqcx/xpw9LIbcE4VlYZQO5rrS34+9I=";
   };
 
   vendorSha256 = null;
 
-  doCheck = false;
-
   ldflags = [ "-X main.Version=${version}" ];
 
+  preCheck = ''
+    # Workaround for: sshgen_test.go:74:
+    #   mkdir /homeless-shelter/.cloudflared: no such file or directory
+
+    export HOME="''$(mktemp -d)";
```

```suggestion
    # Workaround for: sshgen_test.go:74: mkdir /homeless-shelter/.cloudflared: no such file or directory
    export HOME="$(mktemp -d)";
```
thats pretty common.